android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -195,12 +195,12 @@ open class QuestionnaireActivity : BaseMultiLanguageActivity(), View.OnClickList
       this,
       { result ->
         if (result) {
+          saveProcessingAlertDialog.dismiss()
           finish()
         } else {
+          saveProcessingAlertDialog.dismiss()
           Timber.e("An error occurred during extraction")
         }
```

Comment:
I'm not sure what the correct syntax is, but can we reduce duplication by moving that new line above the conditional?

maybe it needs to be wrapped in another block for syntax reasons

```suggestion
        saveProcessingAlertDialog.dismiss()
        if (result) {
          finish()
        } else {
          Timber.e("An error occurred during extraction")
        }
```